### PR TITLE
2018-01-26 Gaeul : code for finding opening cafe

### PR DESCRIPTION
**Pull request by GaeulG**

### CODE REVIEW

File: server/findingCafe.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import json
+import pymysql
+from datetime import datetime
+
+open_cafe_list = []
+
+# connect to DB
+def getConnection():
+    return pymysql.connect(host='localhost',
+    user='gaeul',
```
Comment (shberry) [inline server/findingCafe.py]: indentation

File: server/findingCafe.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    now_minute = now.minute
+    return now_day, now_hour, now_minute #return type is tuple 
+
+# find out opening cafe
+def findOpenCafes(cafes):
+    now_day, now_hour, now_minute = getDate()
+    day_to_find = now_day + 7 # the number '7' is to match with attribute 
+    #loop for each cafe in cafe list
+    for cafe in cafes :
+        print (cafe)
```
Comment (shberry) [inline server/findingCafe.py]: Is this for testing? If not, we shouldn't include this in production code

File: server/findingCafe.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            closing_hour = cafe[day_to_find][2]
+            open_hour = int(cafe[day_to_find][6] + cafe[day_to_find][7])
+            open_minute = int(cafe[day_to_find][8] + cafe[day_to_find][9])
+            close_hour = int(cafe[day_to_find][15] + cafe[day_to_find][16])
+            close_minute = int(cafe[day_to_find][17] + cafe[day_to_find][18])
+            if (((open_hour == now_hour and open_minute <= now_minute) or (open_hour < now_hour)) and \
+                ((close_hour == now_hour and close_minute >= now_minute) or (close_hour > now_hour))) :
+                open_cafe_list.append(cafe[1])
+    return open_cafe_list
+
```
Comment (shberry) [inline server/findingCafe.py:58]: remove extra lines

File: server/findingCafe.py (hunk 2 above)
Comment (shberry) [reply server/findingCafe.py]: Same applies to all print statements.

File: server/findingCafe.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+
+if __name__ == "__main__":
+    cafe_list = getTable()
```
Comment (shberry) [inline server/findingCafe.py]: Is there a reason we're pulling up the whole table? Can we improve the query and retrieve only the information we need? (If the table gets big, select * is a bad way to go and things become super slow)

**Review by shberry — COMMENTED** [review]
Left a few comments. Thanks!

File: server/findingCafe.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    day_to_find = now_day + 7 # the number '7' is to match with attribute 
+    #loop for each cafe in cafe list
+    for cafe in cafes :
+        print (cafe)
+        print (cafe[1],cafe[day_to_find])
+        if cafe[day_to_find] == 'NULL' :
+            continue
+        else :
+            open_hour = cafe[day_to_find][1]
+            closing_hour = cafe[day_to_find][2]
```
Comment (khc196) [inline server/findingCafe.py:49]: What do these variables mean?

File: server/findingCafe.py (hunk 1 above)
Comment (GaeulG) [reply server/findingCafe.py]: I changed !

File: server/findingCafe.py (hunk 2 above)
Comment (GaeulG) [reply server/findingCafe.py]: I deleted all

File: server/findingCafe.py (hunk 3 above)
Comment (GaeulG) [reply server/findingCafe.py:58]: okay!

File: server/findingCafe.py (hunk 4 above)
Comment (GaeulG) [reply server/findingCafe.py]: It is just for test and it was added to hwancheol's code.
Hwancheol makes query so I don't need to make query in this code :)

File: server/findingCafe.py (hunk 5 above)
Comment (GaeulG) [reply server/findingCafe.py:49]: I think i should have delete these code.
It was a test for my code:) thank u for asking

**Review by khc196 — APPROVED** [review]
Cool.

**Pull request merged**